src/publish/s3.js

```diff
@@ -3,9 +3,22 @@ const winston = require("winston-color")
 const zlib = require("zlib")
 const config = require("../config")
 
-const S3 = new AWS.S3()
+let conf = {}
 
+// This is the case where using custom s3 api-like services like minio.
+if (config.aws.endpoint) {
+  conf = {
+    accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+    secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+    endpoint: (config.aws.endpoint ? config.aws.endpoint : ''),
+    s3ForcePathStyle: 'true',
+    signatureVersion: 'v4'
+  }
```

Comment:
ternary operator removed.
I don't know if `s3ForcePathStyle: 'true'` and `signatureVersion: 'v4'` could hurt the standard S3 servers, they are taken from the minio docs: https://docs.minio.io/docs/how-to-use-aws-sdk-for-javascript-with-minio-server